test/NuGet.Core.Tests/NuGet.Commands.Test/PackCommandRunnerTests.cs

```diff
@@ -48,6 +48,71 @@ public void RunPackageBuild_WithDefaultExcludes_ExcludesDefaultExcludes()
             }
         }
 
+        [Fact]
+        public void RunPackageBuild_WithInvalidNuspec_Throws()
+        {
+            using (var test = DefaultExclusionsTest.Create(validNuspec: false))
+            {
+                var args = new PackArgs()
+                {
+                    CurrentDirectory = test.CurrentDirectory.FullName,
+                    Exclude = Enumerable.Empty<string>(),
+                    Logger = NullLogger.Instance,
+                    Path = test.NuspecFile.FullName,
+                };
+                var runner = new PackCommandRunner(args, createProjectFactory: null);
+
+                Assert.Throws<Exception>(() => runner.RunPackageBuild());
+
+                var nupkgPath = Path.Combine(test.CurrentDirectory.FullName, $"DefaultExclusions.1.0.0.nupkg");
+                Assert.False(File.Exists(nupkgPath), "The output .nupkg should not exist when pack fails.");
+            }
+        }
+
+        [Fact]
+        public void RunPackageBuild_WithValidNuspec_WithVersionAsArgument_PackSucceeds()
+        {
+            using (var test = DefaultExclusionsTest.Create(validNuspec: true))
+            {
+                var args = new PackArgs()
+                {
+                    CurrentDirectory = test.CurrentDirectory.FullName,
+                    Exclude = Enumerable.Empty<string>(),
+                    Logger = NullLogger.Instance,
+                    Path = test.NuspecFile.FullName,
+                    Version = "3.1.2"
+                };
+                var runner = new PackCommandRunner(args, createProjectFactory: null);
+
+                var result = runner.RunPackageBuild();
+                Assert.True(result);
+
+                var nupkgPath = Path.Combine(test.CurrentDirectory.FullName, $"DefaultExclusions.3.1.2.nupkg");
+                Assert.True(File.Exists(nupkgPath), "nuspec file does not exist");
+            }
+        }
+
+        [Fact]
+        public void RunPackageBuild_WithInvalidNuspec_WithVersionAsArgument_PackSucceeds()
```

Comment:
```suggestion
        public void RunPackageBuild_NuspecWithEmptyVersion_WithVersionAsArgument_PackSucceeds()
```

Describing the test scenario as "invalid nuspec" kind of suggests that using `<package />` should work